omp/preconditioner/jacobi_kernels.cpp

```diff
@@ -118,7 +559,34 @@ void convert_to_dense(
     const Array<IndexType> &block_pointers, const Array<ValueType> &blocks,
     const preconditioner::block_interleaved_storage_scheme<IndexType>
         &storage_scheme,
-    ValueType *result_values, size_type result_stride) NOT_IMPLEMENTED;
+    ValueType *result_values, size_type result_stride)
+{
+    const auto ptrs = block_pointers.get_const_data();
+    const auto prec = block_precisions.get_const_data();
+    const size_type matrix_size = ptrs[num_blocks];
+#pragma omp parallel for
+    for (size_type i = 0; i < matrix_size; ++i) {
+        for (size_type j = 0; j < matrix_size; ++j) {
+            result_values[i * result_stride + j] = zero<ValueType>();
+        }
+    }
+#pragma omp parallel for
+    for (size_type i = 0; i < num_blocks; ++i) {
+        const auto group =
+            blocks.get_const_data() + storage_scheme.get_group_offset(i);
+        const auto block_size = ptrs[i + 1] - ptrs[i];
+        const auto p = prec ? prec[i] : precision_reduction();
+        GKO_PRECONDITIONER_JACOBI_RESOLVE_PRECISION(
+            ValueType, p,
+            transpose_block(
+                block_size,
+                reinterpret_cast<const resolved_precision *>(group) +
+                    storage_scheme.get_block_offset(i),
+                storage_scheme.get_stride(),
+                result_values + ptrs[i] * result_stride + ptrs[i],
+                result_stride));
+    }
+}
```

Comment:
Only now I realize that jacobi kernels were also pushed as part of this MR, but no tests were added for this (and I don't remember reviewing this).

I found that out thanks to the new coverage.
https://my.cdash.org/viewCoverage.php?buildid=1611563

Reply:
ouch, that was not intended. They should come later...